File: exercise-2022/simulation-2016/8.cpp
#include <iostream>
using namespace std;

int quan[] = {7, 9, 10, 5, 8, 4, 2, 1, 6, 3, 7, 9, 10, 5, 8, 4, 2};
char m[] = {'1', '0', 'X', '9', '8', '7', '6', '5', '4', '3', '2'};

int main()
{
    int n, f = 0;
    cin >> n;
    while (n --)
    {
        string s;
        cin >> s;
        long long val = 0, ff = 0;
        for (int i=0; i<17; i++)
        {
            int num = s[i] - '0';
            if (num<0 || num>9)
            {
                ff = 1;
                break;
            }
            val += num * quan[i];
        }
        char code = m[val%11];
        if (ff || s[17] != code)
        {
            cout << s << endl;
            f = 1;
        }
    }
    if (! f) cout << "All passed" << endl;
    return 0;
}

File: exercise-2022/simulation-2016/Python/8.py
def dev(a, b):
    return int(a / b)
def mod(a, b):
    return a - dev(a, b) * b

quan = [7, 9, 10, 5, 8, 4, 2, 1, 6, 3, 7, 9, 10, 5, 8, 4, 2]
zhi = ['1', '0', 'X', '9', '8', '7', '6', '5', '4', '3', '2']
n = int(input())
printed = False
for _ in range(n):
    he = 0
    flag = 0
    shenfen = input()
    for i, c in enumerate(shenfen[0:17]):
        if c.isdigit():
            he += int(c)*quan[i]
        else:
            flag = 1
            break
    if flag or zhi[mod(he, 11)] != shenfen[17]:
        print(shenfen)
        printed = True
if not printed:
    print('All passed')